File: car-repair-shop-backend/shop/src/main/java/car/repair/shop/repair/request/RepairRequest.java
package car.repair.shop.repair.request;

import car.repair.shop.repair.request.controller.dto.SubmitRepairRequestDto;
import lombok.AccessLevel;
import lombok.Getter;

import java.time.LocalDateTime;
import java.util.List;

//todo either private setters for @document or AllArgsConstructor
@Getter(AccessLevel.PACKAGE)
class RepairRequest {
    private RepairRequestId repairRequestId;
    private String vin;
    private String issueDescription;
    private String submitterFirstName;
    private String submitterLastName;
    private String email;
    private String phoneNumber;
    private List<PreferredVisitWindow> preferredVisitWindows;
    private LocalDateTime submittedAt;
    private LocalDateTime handledAt;
    private RepairRequestStatus repairRequestStatus;

    static RepairRequest from(SubmitRepairRequestDto submitRepairRequestDto) {
        var repairRequest = new RepairRequest();
        //todo
        return repairRequest;
    }

    protected void markAsHandled() {
        handledAt = LocalDateTime.now();
        repairRequestStatus = RepairRequestStatus.HANDLED;
    }

    protected void maskAsAppointmentMade() {
        repairRequestStatus = RepairRequestStatus.APPOINTMENT_MADE;
    }
}

File: repair-requests-portal/src/app/models/repair-request.ts
import { PreferredVisitWindowDto } from "./preferred-visit-window";

export interface RepairRequest {
    id: string;
    vin: string;
    issueDescription: string;
    submitterFirstName: string;
    submitterLastName: string;
    email: string;
    phoneNumber: string;
    asap: boolean;
    preferredVisitWindows: PreferredVisitWindowDto[];
    submittedAt: string;
    handledAt: string;
    status: string;
  }